Objects/typeobject.c

```diff
@@ -4650,6 +4641,12 @@ type_new_descriptors(const type_new_ctx *ctx, PyTypeObject *type, PyObject *dict
     if (et->ht_slots != NULL) {
         PyMemberDef *mp = _PyHeapType_GET_MEMBERS(et);
         Py_ssize_t nslot = PyTuple_GET_SIZE(et->ht_slots);
+        if (ctx->base->tp_itemsize != 0) {
+            PyErr_Format(PyExc_TypeError,
+                         "nonempty __slots__ not supported for subtype of '%s'",
```

Comment:
The message is not tecnically correct any more; maybe use this?
```suggestion
                         "arbitrary __slots__ not supported for subtype of '%s'",
```